### ISSUE TITLE
[Question] General recommendation for where to put license file

### ISSUE THREAD

**Issue by xylosper** [conan-io/conan#1316]
I'd like to put license and notification of copyright files in package, so that I can import them later when distributing what depends on them.

Is there any general rule for this purpose? Or, should I make my own rule?

**Comment by memsharded** [comment]
What do you mean by "rule"? Not anything enforced by conan, but IMO the approach is:
- License files should be packaged in the root folder of the package.
- It requires its ``self.copy("License.txt")`` or similar in the ``package()`` method
- From there, an ``imports`` with *.txt pattern, or "License" pattern should work.

**Comment by xylosper** [comment]
What I meant by rule was something like the package rule in Linux distros, for instance, executable should be in `/usr/bin` and libraries should be in `/usr/lib` and so on.

I've seen that a conan pacakge contains similar folders(`bin`, `lib`, `include`), I guessed something like `share/license/[package-name]`.

If I put license file on root folder, how can I distinguish license files for each package?
Lots of projects use same or similar file name for license, `license` or `copying`.
If I just imported them from root, it would cause confiliction in many cases, and, if I were lucky, I could get some several files named license.txt, license.md, etc and it would be very difficult to identify that which package follows which license.

Most of opensource libraries require notification of their license terms, and I hoped that maybe I can use conan for such purpose.

For instance, I can put a license file for `package-a` into `share/license/package-a` and then by importing `share/lincese`, I can make a directory `licenses` such as,
```
/ (project root where to run conan install)
...
/licenses/pacakge-a/LICENSE
/licenses/pacakge-b/LICENSE.md
/licenses/pacakge-c/COPYING.md
...
```

Before I make this move, I just wanted to know if there's already any rule for that, but from your answer, I think it wasn't considered. Thank you.

**Comment by memsharded** [comment]
Oh, I see, now I understand, this is a valid concern. I think this issue might need some polishing, maybe just in the docs, or consider such a rule, or a better example how to improve it. Lets re-open and will review it for 0.24. Thanks very much for the feedback.

**Comment by xylosper** [comment]
Thank you for your consideration. The ultimate purpose is to provide some kind of license notification for dependencies.

If you consider to make a new command or option for this purpose, it would be better that conan can generate an information file such as what you can see here: https://github.com/mongodb/mongo-cxx-driver/blob/releases/stable/THIRD-PARTY-NOTICES .

Although conan lacks lots of thing to generate such details like copyright holder, I think conan can generate a simple file which contains a list of which each item refers license file/license name for each pacakge, for instance:

```
Copyright notifitcation

This software uses other libraries or resources which should be distributed by their own licenses.

* package-a (<license>)
See <license file path>

* package-b (<license>)
See <license file path>

...
```

This file should be generate while importing/copying license file into project folder.
I'm not sure it is enough to refer only one file. Maybe it would be better to refer several files or a directory which contains such files, because some projects may consist of different license parts, or, they may allow to choose license from multiple options. For instance, Qt provides triple licenses, which contain two opensource licesnses(GPL and LGPL) and one commercial license.

...and, yes, I agree with that this feature should be polished a lot more.

**Comment by memsharded** [comment]
Right  now, it is possible to gather the "declared" licenses for dependencies with the ``conan info`` command. But that is different from actually collecting all possible "license" files for the final "product".
I am not sure about the automatic generation of the summary, but definitely sound a real use-case, so thanks for the suggestion, we will take it into account.

**Comment by memsharded** [comment]
This has been merged. It still has to be documented, but it will be in the next release. In the  meantime, you can check in the test in https://github.com/conan-io/conan/pull/1327/files#diff-fe508c3b10cf5e8e2f8c383e9af96307, that adding the ``folder`` and ``ignore_case`` arguments, the output will be a set of folders, each one with its license.

**Comment by memsharded** [comment]
Functionality implemented and released in 0.24.
Pending docs: https://github.com/conan-io/docs/issues/229
